src/pip/_internal/utils/temp_dir.py

```diff
@@ -98,7 +98,7 @@ class AdjacentTempDirectory(TempDirectory):
 
     """
     # The characters that may be used to name the temp directory
-    LEADING_CHARS = "-~.+=%0123456789"
+    LEADING_CHARS = "~.=%0123456789"
```

Comment:
Since the reason isn't obvious and for anyone in the future that might want to modify this, I think it would be good to add a comment here recording the lesson of why `-` isn't in the list (see https://github.com/pypa/pip/issues/6194#issuecomment-459117736).

It's okay for this to be done in PR #6215 after this is merged.

Reply:
Sure, will do